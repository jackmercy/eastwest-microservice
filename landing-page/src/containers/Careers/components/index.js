export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
