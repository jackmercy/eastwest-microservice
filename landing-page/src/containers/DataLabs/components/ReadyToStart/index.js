import dynamic from 'next/dynamic';

export default dynamic(() => import('./ReadyToStart'), { ssr: false });
