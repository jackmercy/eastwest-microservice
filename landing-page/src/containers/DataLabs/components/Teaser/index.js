import dynamic from 'next/dynamic';

export default dynamic(() => import('./Teaser'), { ssr: false });
