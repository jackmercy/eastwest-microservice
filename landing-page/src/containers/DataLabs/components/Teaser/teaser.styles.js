import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    height: '100vh',
    boxSizing: 'border-box',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'left',
    overflow: 'hidden',
  },
  contentLayout: { position: 'relative', height: '100%' },
  content: {
    top: '58%',
    right: 0,
    maxWidth: rem(700),
    transform: 'translateY(-50%)',
    position: 'absolute',
    borderRadius: rem(12),
    color: COLORS.textWhite,
    padding: [rem(40), rem(32), rem(48)],
    boxSizing: 'border-box',
    transition: 'padding .5s ease, left .5s ease, max-width .5s ease .3s',
    backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.75),
  },
  logo: { maxWidth: '100%' },
  purpose: {
    fontWeight: 900,
    fontSize: rem(24),
    fontStyle: 'italic',
    textAlign: 'right',
    lineHeight: rem(32),
    margin: 0,
    transition: 'font-size .5s ease',
  },
  description: {
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    marginTop: rem(24),
    marginBottom: 0,
    transition: 'font-size .5s ease',
  },
  buttonGroup: { marginTop: rem(32) },
  button: {
    padding: [0, rem(32)],
    '& + $button': { marginLeft: rem(24) },
    '&.tertiary': {
      color: COLORS.textWhite,
      '&:not(.disabled):hover': { color: COLORS.primary },
      '& > [class^=icon-]': { fontSize: rem(10), marginLeft: rem(16) },
    },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxLg}px)`]: {
    content: {
      maxWidth: '100%',
      width: '100%',
      boxSizing: 'border-box',
      position: 'relative',
    },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    purpose: { fontSize: rem(14), lineHeight: rem(16) },
    content: { padding: [rem(24), rem(16), rem(30)] },
    description: { fontSize: rem(16), lineHeight: rem(24), marginTop: rem(16) },
    buttonGroup: { marginTop: rem(24) },
    button: {
      display: 'flex',
      margin: [0, 'auto'],
      '& + $button': { margin: [rem(24), 'auto', 0] },
    },
  },
});
