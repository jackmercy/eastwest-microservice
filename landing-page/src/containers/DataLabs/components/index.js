export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
