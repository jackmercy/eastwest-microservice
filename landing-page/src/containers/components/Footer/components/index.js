export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
