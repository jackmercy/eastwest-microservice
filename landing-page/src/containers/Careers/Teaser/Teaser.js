import { memo, useCallback } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { Button } from '../../../components';
import { Content } from '../../components';
import { scrollToElement } from '../../../utils';

import useStyles from './teaser.styles';

const Teaser = ({ CTAId }) => {
  const intl = useIntl();
  const classes = useStyles();

  const handleCTAClick = useCallback(() => {
    scrollToElement(CTAId);
  }, [CTAId]);

  return (
    <>
      <Content className={classes.root}>
        <h1 className={classes.title} data-test="teaser-title">
          {intl.formatMessage(
            { id: 'CAREERS.TEASER.TITLE' },
            {
              span: (...chunks) => (
                <span className={classes.primary}>{chunks.join('')}</span>
              ),
            },
          )}
        </h1>
        <p className={classes.description} data-test="teaser-description">
          {intl.formatMessage({ id: 'CAREERS.TEASER.DESCRIPTION' })}
        </p>
        <Button
          color="primary"
          className={classes.button}
          dataTest="teaser-cta"
          onClick={handleCTAClick}
        >
          {intl.formatMessage({ id: 'CAREERS.TEASER.BUTTON' })}
        </Button>
      </Content>
      <div className={classes.image} data-test="teaser-image" />
    </>
  );
};
Teaser.propTypes = {
  CTAId: PropTypes.string.isRequired,
};

export default memo(Teaser);
