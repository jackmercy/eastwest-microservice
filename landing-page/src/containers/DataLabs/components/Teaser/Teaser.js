import Link from 'next/link';
import { useRouter } from 'next/router';
import { memo, useCallback } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { useMediaQuery } from 'react-responsive';

import { Content } from '../../../components';
import { Button, Logo } from '../../../../components';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

import useStyles from './teaser.styles';

const Teaser = ({
  image,
  mobileImage,
  purpose,
  description,
  otherText,
  isBrands,
  defaultReason,
}) => {
  const intl = useIntl();
  const router = useRouter();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxSm}px)`,
  });

  const handleMoreClick = useCallback(() => {
    router.push(
      { pathname: window.location.pathname, query: { isBrands: !isBrands } },
      window.location.pathname,
    );
  }, [router, isBrands]);

  return (
    <div
      className={classes.root}
      style={{ backgroundImage: `url(${isMobile ? mobileImage : image})` }}
      data-test="teaser-cover-image"
    >
      <Content className={classes.contentLayout}>
        <div className={classes.content}>
          <Logo type="data-labs" className={classes.logo} />
          <p className={classes.purpose} data-test="teaser-purpose">
            {purpose}
          </p>
          <p className={classes.description} data-test="teaser-description">
            {description}
          </p>
          <div className={classes.buttonGroup}>
            <Link
              href={`/contact/?defaultReason=${defaultReason}#contact-form`}
              passHref
              as="/contact/#contact-form"
            >
              <Button
                tag="a"
                size="large"
                color="primary"
                keepSizeOnMobile
                className={classes.button}
                dataTest="teaser-button-get-started"
              >
                {intl.formatMessage({ id: 'DATA_LABS.TEASER.BUTTON' })}
              </Button>
            </Link>
            <Button
              tag="a"
              size="large"
              color="tertiary"
              keepSizeOnMobile
              onClick={handleMoreClick}
              className={classes.button}
              dataTest="teaser-button-more"
            >
              {otherText}
              <i className="icon-arrow-right" />
            </Button>
          </div>
        </div>
      </Content>
    </div>
  );
};
Teaser.propTypes = {
  image: PropTypes.string.isRequired,
  mobileImage: PropTypes.string.isRequired,
  purpose: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  otherText: PropTypes.string.isRequired,
  defaultReason: PropTypes.number.isRequired,
  isBrands: PropTypes.bool,
};
Teaser.defaultProps = {
  isBrands: false,
};

export default memo(Teaser);
