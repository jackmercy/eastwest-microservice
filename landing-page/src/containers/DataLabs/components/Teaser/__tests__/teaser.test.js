import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Logo } from '../../../../../components';
import Teaser from '../Teaser';

describe('Testing DataLabs components > Teaser', () => {
  let wrapper;
  const props = {
    image: '/static/images/data-labs/influencers/teaser.png',
    mobileImage: '/static/images/data-labs/influencers/teaser-mobile.png',
    purpose: 'nothing',
    description: 'something',
    otherText: 'more',
    defaultReason: 5,
  };

  beforeAll(() => {
    wrapper = mount(<Teaser {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Teaser {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a cover image', () => {
    expect(wrapper.find('[data-test="teaser-cover-image"]')).toHaveLength(1);
  });

  it('should have a logo', () => {
    expect(wrapper.find(Logo)).toHaveLength(1);
  });

  it('should have a purpose of teaser', () => {
    expect(wrapper.find('[data-test="teaser-purpose"]')).toHaveLength(1);
  });

  it('should have a description', () => {
    expect(wrapper.find('[data-test="teaser-description"]')).toHaveLength(1);
  });

  it('should have a button get started', () => {
    expect(
      wrapper.find('[data-test="teaser-button-get-started"]'),
    ).toHaveLength(1);
  });

  it('should have a button learn more', () => {
    expect(wrapper.find('[data-test="teaser-button-more"]')).toHaveLength(1);
  });
});
