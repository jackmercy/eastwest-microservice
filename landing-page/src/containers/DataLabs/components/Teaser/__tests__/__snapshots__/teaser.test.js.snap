// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataLabs components > Teaser should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test="teaser-cover-image"
  style={
    Object {
      "backgroundImage": "url(/static/images/data-labs/influencers/teaser.png)",
    }
  }
>
  <Memo(Content)
    className="contentLayout-0-2-2"
  >
    <div
      className="content-0-2-3"
    >
      <Memo(Logo)
        className="logo-0-2-4"
        type="data-labs"
      />
      <p
        className="purpose-0-2-5"
        data-test="teaser-purpose"
      >
        nothing
      </p>
      <p
        className="description-0-2-6"
        data-test="teaser-description"
      >
        something
      </p>
      <div
        className="buttonGroup-0-2-7"
      >
        <Link
          as="/contact/#contact-form"
          href="/contact/?defaultReason=5#contact-form"
          passHref={true}
        >
          <Memo([object Object])
            className="button-0-2-8"
            color="primary"
            dataTest="teaser-button-get-started"
            keepSizeOnMobile={true}
            size="large"
            tag="a"
          >
            Get started
          </Memo([object Object])>
        </Link>
        <Memo([object Object])
          className="button-0-2-8"
          color="tertiary"
          dataTest="teaser-button-more"
          keepSizeOnMobile={true}
          onClick={[Function]}
          size="large"
          tag="a"
        >
          more
          <i
            className="icon-arrow-right"
          />
        </Memo([object Object])>
      </div>
    </div>
  </Memo(Content)>
</div>
`;
